enclave-runtime/src/lib.rs

```diff
@@ -285,53 +280,13 @@ pub unsafe extern "C" fn mock_register_enclave_xt(
 	)
 	.encode();
 
-	*nonce += 1;
+	*nonce_lock = Nonce(nonce_value + 1);
+	std::mem::drop(nonce_lock);
```

Comment:
just a small optimization: we drop the nonce write lock early, to minimize the time we're holding the lock.